prospero-cli/src/test/java/org/wildfly/prospero/cli/commands/ChannelCommandTest.java

```diff
@@ -277,6 +283,59 @@ public void testRemoveEmptyChannel() {
+        int exitCode = commandLine.execute(CliConstants.Commands.CHANNEL, CliConstants.Commands.ADD,
+                CliConstants.DIR, dir.toString(),
+                CliConstants.CHANNEL_NAME, "channel-0",
+                CliConstants.CHANNEL, path.toString());
+        Assert.assertEquals(ReturnCodes.SUCCESS, exitCode);
+
+        InstallationMetadata installationMetadata = InstallationMetadata.loadInstallation(dir);
+        List<Channel> registeredChannels = installationMetadata.getProsperoConfig().getChannels();
+        Channel addedChannel = registeredChannels.get(3);
+        Assert.assertEquals(addedChannel.getName(), "channel-0");
+        assertThat(addedChannel.getRepositories().stream()
+                .map(r -> Tuple.tuple(r.getId(), r.getUrl()))
+                .collect(Collectors.toList())) // Convert the Stream to a List
+                .containsExactly(
+                        Tuple.tuple("maven-central", "https://repo1.maven.org/maven2/"),
+                        Tuple.tuple("nexus", "https://repository.jboss.org/nexus/content/groups/public"),
+                        Tuple.tuple("maven-redhat-ga", "https://maven.repository.redhat.com/ga"));
+    }
+    @Test
+    public void testChannelAddInvalidParameters() throws IOException {
+        Path path = MetadataTestUtils.prepareChannel("manifest.yaml");
+        int exitCode = commandLine.execute(CliConstants.Commands.CHANNEL, CliConstants.Commands.ADD,
+                CliConstants.DIR, dir.toString(),
+                CliConstants.CHANNEL_NAME, "channel-0",
+                CliConstants.CHANNEL, path.toString(),
+                CliConstants.CHANNEL_MANIFEST, "org.test:test",
+                CliConstants.REPOSITORIES, "test_repo::http://test.te");
+        Assert.assertEquals(ReturnCodes.INVALID_ARGUMENTS, exitCode);
+
+        exitCode = commandLine.execute(CliConstants.Commands.CHANNEL, CliConstants.Commands.ADD,
+                CliConstants.DIR, dir.toString(),
+                CliConstants.CHANNEL_NAME, "channel-1");
+        Assert.assertEquals(ReturnCodes.INVALID_ARGUMENTS, exitCode);
+
+        File InvalidChannelFile = tempDir.newFile("InvalidChannelFile.yaml");
+        List<String> yamlContent = List.of(
+                "channels:",
+                "  - name: invalid-channel",
+                "    description: This is an invalid channel for testing purposes.",
+                "    repositories:",
+                "      - id: test-repo",
+                "        url: http://invalid-url.org/repo"
+        );
+        Files.write(InvalidChannelFile.toPath(), yamlContent);
+        exitCode = commandLine.execute(CliConstants.Commands.CHANNEL, CliConstants.Commands.ADD,
+                CliConstants.DIR, dir.toString(),
+                CliConstants.CHANNEL_NAME, "channel-2",
+                CliConstants.CHANNEL, InvalidChannelFile.getAbsolutePath());
+        Assert.assertEquals(ReturnCodes.PROCESSING_ERROR, exitCode);
```

Comment:
@spyrkob I am not sure why exit code is coming 2 here. Could you please take a look?
Thanks

Reply:
@parsharma do you mean why it's returning PROCESSING_ERROR? Either code is fine in this case. Generally INVALID_ARGUMENT is related to syntax -e.g. missing argument, etc while PROCESSING_ERROR refers to issue executing command.

However when running this example in command line I get:
```
org.wildfly.channel.InvalidChannelMetadataException: Invalid Manifest
caused by:
 The manifest does not specify a schemaVersion.

	at org.wildfly.channel.ChannelMapper.getSchema(ChannelMapper.java:80)
	at org.wildfly.channel.ChannelMapper.validateString(ChannelMapper.java:146)
	at org.wildfly.channel.ChannelMapper.fromString(ChannelMapper.java:125)
	at org.wildfly.prospero.cli.commands.channel.ChannelAddCommand.call(ChannelAddCommand.java:80)
	at org.wildfly.prospero.cli.commands.channel.ChannelAddCommand.call(ChannelAddCommand.java:42)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2045)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2465)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2457)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2419)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
	at picocli.CommandLine.execute(CommandLine.java:2174)
	at org.wildfly.prospero.cli.CliMain.execute(CliMain.java:132)
	at org.wildfly.prospero.cli.CliMain.main(CliMain.java:60)
```

The program shouldn't show exception and stack trace to the user, rather a message describing the problem. `InstallCommand` handles similar case for `--channel` argument